ipatests/test_integration/test_trust.py

```diff
@@ -268,6 +259,65 @@ def test_override_homedir(self):
             tasks.restore_files(self.master)
             tasks.clear_sssd_cache(self.master)
 
+    def test_extdom_plugin(self):
+        """Extdom plugin should not return error (32)/'No such object'
+
+        Regression test for https://pagure.io/freeipa/issue/8044
+
+        If there is a timeout during a request to SSSD the extdom plugin
+        should not return error 'No such object' and the existing user should
+        not be added to negative cache on the client.
+        """
+        client = self.clients[0]
+        tasks.backup_file(self.master, paths.SSSD_CONF)
+        log_file = '{0}/sssd_{1}.log' .format(paths.VAR_LOG_SSSD_DIR,
+                                              self.master.domain.name)
+        logsize = len(self.master.get_file_contents(log_file))
+        res = self.master.run_command(['pidof', 'sssd_be'])
+        pid = res.stdout_text.strip()
+        try:
+            test_id = 'id testuser@%s' % self.ad_domain
+            self.master.run_command(test_id)
+            tasks.kinit_admin(self.master)
+            conn = self.master.ldap_connect()
+            entry = conn.get_entry(self.dn)  # pylint: disable=no-member
+            # set the extdom plugin timeout to 1s (1000)
+            entry.single_value['ipaextdommaxnsstimeout'] = 1000
+            conn.update_entry(entry)  # pylint: disable=no-member
+            self.master.run_command(['ipactl', 'restart'])
+            domain = self.master.domain
+            tasks.modify_sssd_conf(
+                self.master,
+                domain.name,
+                {
+                    'timeout': '999999'
+                }
+            )
+            client.run_command(test_id)
+            remove_cache = 'sss_cache -E'
+            self.master.run_command(remove_cache)
+            client.run_command(remove_cache)
+            # stop sssd_be, needed to simulate a timeout in the extdom plugin.
+            stop_sssd_be = self.master.run_command('kill -STOP %s' % pid)
+            client.run_command(test_id)
+
+            error = 'ldap_extended_operation result: No such object(32)'
+            sssd_log2 = self.master.get_file_contents(log_file)[logsize:]
+            assert error.encode() not in sssd_log2
+        finally:
+            if stop_sssd_be.returncode == 0:
+                self.master.run_command('kill -CONT %s' % pid)
+            # set back to default extdom plugin
+            entry_ldif = textwrap.dedent("""
```

Comment:
The code will look cleaner and more consistent if you use same approach here as in test setup (get_entry/update_entry). 
Another question: previously you were updating field ipaExtdomMaxNssTimeout with value 10000 here. Now you delete it.Can you please explain why you changed it?

Reply:
Yes as per discussion with @sumit-bose  to set it default time-out this approach can be also used .

Reply:
The test cleanup should revert the system to original state, not to "original-equivalent". We want to minimize influence of this test on the following ones. Even if it is equivalent now it does not mean it will be always so - due to regression or intentional change. So if this value is defined before the test, we should keep it an restore here. If it is not defined before the test, we should remove it.